analytics/adc/basereader.py

```diff
@@ -37,9 +43,10 @@ def start_reading(self):
     def _read_adc(self, channel) -> Generator[float, None, None]:
         None
 
+    @exponential_backoff_decorator(base_delay_in_seconds=0.1, logger=logger)
     def get_current_buffers(self):
         """Get current state of buffers, converted to an numpy Array"""
-        inner = self.inner_buf 
-        outer = self.outer_buf
+        inner, outer = self.inner_buf, self.outer_buf
+        if (not inner.is_full() or not outer.is_full()):
+            raise BufferNotFull("One of the buffers is not full yet")
```

Comment:
Sorry to be picky haha
```suggestion
            raise BufferNotFull("At least one of the buffers is not full yet")
```